src/python/pants/backend/native/subsystems/native_toolchain.py

```diff
@@ -219,7 +260,7 @@ def select_gcc_c_toolchain(platform, native_toolchain):
     # TODO: we should be providing all of these (so we can eventually phase out XCodeCLITools
     # entirely).
     xcode_clang = yield Get(CCompiler, XCodeCLITools, native_toolchain._xcode_cli_tools)
-    new_include_dirs = xcode_clang.include_dirs + provided_gcc.include_dirs
+    new_include_dirs = provided_gcc.include_dirs + xcode_clang.include_dirs
```

Comment:
Why was this swap necessary?

Reply:
This is actually a correction which is necessary for tests to pass (this is due to the gcc headers' use of `#include_next`). The same switch is done for the `GCCCppToolchain`.

Reply:
...and so this probably deserves a comment in the code. Your cohort which is probably the closest to this code found it mysterious - excellent indicator that it truly is and should get note not just in review.

Reply:
I agree. It has historically been difficult to justify spending time on fixing tech debt for this backend as I wasn't able to get a lot of help understanding the time cost of different approaches, so I attempted to muddle through solutions and leave millions of TODOs because I was extremely concerned about wasting time on unnecessary abstractions. @CMLivingston has been extremely helpful in developing the larger picture. The first longstanding issue is this PR, which addresses the lack of toolchain choice, which you have mentioned multiple times previously (which was appreciated). The second is #6855, which I have listed in a couple comments in this file, and which is intended to remove the mystery from repeated but slightly varying invocations, and to remove the chance of making ordering mistakes like the above by combining `Executable`s at a higher level of abstraction. The last is #5970, which needs to be redone entirely but may be much simpler after the previous are complete.